bedrock/careers/templates/careers/listings.html

```diff
@@ -56,10 +56,15 @@ <h2>Listings</h2>
     </thead>
     <tbody>
     {% for position in positions %}
+      {#
+        When the options list for the Django Form is built, we strip()
+        whitespace from around each option. We need to do the same here
+        to make sure the values can match, else we get filtering 'misses'.
+      #}
       <tr class="position"
-          data-team="{{ position.department }},"
-          data-type="{{ position.position_type }},"
-          data-location="{{ position.location }},">
+          data-team="{{ position.department.strip() }},"
+          data-type="{{ position.position_type.strip() }},"
+          data-location="{{ position.location.strip() }},">
```

Comment:
Looking at [the model](https://github.com/mozilla/bedrock/blob/master/bedrock/careers/models.py#L12), none of these is nullable, so they should all return at least an empty string, which means that `.strip()` is safe to call